azavea/ashlar-schema-editor
# Update development scripts to run a sample server

# Overview

Up until now, I've been developing the schema editor by making changes in the [demo app](https://github.com/jeancochrane/philly-fliers) and then contributing those changes back upstream with `git subtree`. This pattern worked fine for the fellowship, but it's not a sustainable way for other people to contribute to the project.

This PR beefs up the development environment to install and configure a Grout Server instance with sample data so that a developer can run the app locally.

## Notes

- I'm not 100% pleased with my approach to installing the Grout Server instance. I didn't want to create direct coupling between the two apps, so I decided to set up `scripts/update` to install and configure `grout-server` from GitHub. This works, but it has the downside that the `docker-compose.yml` file is technically not valid until the user runs `scripts/update` since the `./grout-server` directory doesn't exist -- not a dealbreaker, but not very clean either. If you have better ways of thinking about this problem, I'm all ears.

## Testing instructions

- Nothing has changed that should affect unit tests, and Travis will run them for you. However, if you'd like to run them locally anyway, run `./scripts/update` and then `./scripts/test`.

- To test that the development server works as expected, run `./scripts/update` and then `./scripts/server` to run the development server. (Note that `./scripts/update` will install `grout-server` from GitHub and will use HTTPS by default; to use SSH instead, call `./scripts/update --ssh`.) Navigate to `localhost:9000` and confirm that everything works as expected.
